packages/browser/src/Session.ts

```diff
@@ -132,16 +135,37 @@ function isLoggedIn(
 /**
  * A {@link Session} object represents a user's session on an application. The session holds state, as it stores information enabling acces to private resources after login for instance.
  */
-export class Session extends EventEmitter {
+export class Session extends EventEmitter implements IHasSessionEventListener {
   /**
    * Information regarding the current session.
    */
   public readonly info: ISessionInfo;
 
+  /**
+   * Session attribute exposing the EventEmitter interface, to listen on session
+   * events such as login, logout, etc.
+   */
+  public readonly events: ISessionEventListener;
+
   private clientAuthentication: ClientAuthentication;
 
   private tokenRequestInProgress = false;
 
+  /**
+   * Proxy handler to ease transition to the Session no longer being an EvenEmitter,
+   * and instead Session["events"] being a SessionEventEmitter. For the time being,
+   * Session["events"] is just a proxy to the SessionEventEmitter interface of
+   * the Session.
+   *
+   * When Session no longer implements SessionEventEmitter, this can be removed.
+   * @hidden
+   */
+  private handler = buildProxyHandler(
+    EventEmitter,
+    Session.prototype,
+    "events only implements SessionEventEmitter"
+  );
```

Comment:
I still wouldn't provide this handler as a property of this class as all, instead it can be created as a constant in the constructor.